Content.Server/Changeling/ChangelingSystem.cs

```diff
@@ -0,0 +1,1114 @@
+    {
+        if (!TryComp<StoreComponent>(uid, out var store))
+            return;
+
+        _store.ToggleUi(uid, uid, store);
+    }
+
+    private void OnAbsorb(EntityUid uid, ChangelingComponent comp, ref AbsorbDNAEvent args)
+    {
+        var target = args.Target;
+
+        if (!IsIncapacitated(target))
+        {
+            _popup.PopupEntity(Loc.GetString("changeling-absorb-fail-incapacitated"), uid, uid);
+            return;
+        }
+        if (HasComp<AbsorbedComponent>(target))
+        {
+            _popup.PopupEntity(Loc.GetString("changeling-absorb-fail-absorbed"), uid, uid);
+            return;
+        }
+        if (!HasComp<AbsorbableComponent>(target))
+        {
+            _popup.PopupEntity(Loc.GetString("changeling-absorb-fail-unabsorbable"), uid, uid);
+            return;
+        }
+
+        if (!TryUseAbility(uid, comp, args))
+            return;
+
+        var popupOthers = Loc.GetString("changeling-absorb-start", ("user", Identity.Entity(uid, EntityManager)), ("target", Identity.Entity(target, EntityManager)));
+        _popup.PopupEntity(popupOthers, uid, PopupType.LargeCaution);
+        PlayMeatySound(uid, comp);
+        var dargs = new DoAfterArgs(EntityManager, uid, TimeSpan.FromSeconds(15), new AbsorbDNADoAfterEvent(), uid, target)
+        {
+            DistanceThreshold = 1.5f,
+            BreakOnDamage = true,
+            BreakOnHandChange = false,
+            BreakOnMove = true,
+            BreakOnWeightlessMove = true,
+            AttemptFrequency = AttemptFrequency.StartAndEnd
+        };
+        _doAfter.TryStartDoAfter(dargs);
+    }
+    public ProtoId<DamageGroupPrototype> AbsorbedDamageGroup = "Genetic";
+    private void OnAbsorbDoAfter(EntityUid uid, ChangelingComponent comp, ref AbsorbDNADoAfterEvent args)
+    {
+        if (args.Args.Target == null)
+            return;
+
+        var target = args.Args.Target.Value;
+
+        PlayMeatySound(args.User, comp);
+
+        if (args.Cancelled || !IsIncapacitated(target) || HasComp<AbsorbedComponent>(target))
+            return;
+
+        var dmg = new DamageSpecifier(_proto.Index(AbsorbedDamageGroup), 200);
+        _damage.TryChangeDamage(target, dmg, false, false);
+        _blood.ChangeBloodReagent(target, "FerrochromicAcid");
+        _blood.SpillAllSolutions(target);
+
+        EnsureComp<AbsorbedComponent>(target);
+
+        var popup = Loc.GetString("changeling-absorb-end-self-ling");
+        var bonusChemicals = 0;
+        var bonusEvolutionPoints = 0;
+        if (HasComp<ChangelingComponent>(target))
+        {
+            bonusChemicals += 60;
+            bonusEvolutionPoints += 10;
+        }
+        else
+        {
+            popup = Loc.GetString("changeling-absorb-end-self", ("target", Identity.Entity(target, EntityManager)));
+            bonusChemicals += 10;
+            bonusEvolutionPoints += 2;
```

Comment:
Changelings should be able to refresh their store and lose all abilities to select other, not just farm evo points by commiting mass genocide